networkit/cpp/graph/GraphTools.cpp

```diff
@@ -200,15 +200,66 @@ double density(const Graph &G) noexcept {
     return m / (n * (n - 1));
 }
 
-double volume(const Graph &G) {
-    if (G.isDirected())
-        throw std::runtime_error("Volume computation is only supported for undirected graphs.");
+double inVolume(const Graph &G, const std::unordered_set<node> &nodes) {
+    double inVolume = 0.0;
+    bool directed = G.isDirected() ? false : true;
+
+    // Compute volume of subgraph
+    if (!nodes.empty() && G.numberOfNodes() != nodes.size()) {
+        for (auto it = std::begin(nodes); it != std::end(nodes); ++it) {
+            inVolume += G.weightedDegreeIn(*it, directed);
+
+            // This removes all incoming edges from the rest of the graph to the subgraph
+            G.forInNeighborsOf(*it, [&](node v) {
+                std::unordered_set<node>::const_iterator neighborIt = nodes.find(v);
+                if (neighborIt == nodes.end()) {
+                    inVolume -= static_cast<double>(G.weight(v, *it));
```

Comment:
The edge weight should instead be taken as parameter in this lambda by defining it as `[&](node, node v, edgeweight w)`.